### PR TITLE
[doc] Extensions: translate various sections from French

### PR DESCRIPTION
**Pull request by Frenzie**
See https://github.com/FreshRSS/FreshRSS/pull/1697#discussion_r154290857

The file is slightly larger than I anticipated, so I won't be able to finish it in one "breakfast translation". Here are this morning's results. I can keep adding to this PR or we can split it into multiple PRs; whichever you prefer.

The quality is first draft, which probably suffices but it means that I might've unintentionally copied French sentence structures or ways of expression.

### CODE REVIEW

**Comment by kevinpapst** [comment]
Could you please change one (quite important) character in the sentence 

> `registerController($base_name)` will tell Minz to take into account the given controller in the routing system. The controller must be located in your `Controllers` directory, the name of the file must be` <base_name>Controller.php` and the name of the `FreshExtension_<base_name>_Controller` class.

`controllers` must be written in lowercase! Compare with https://github.com/FreshRSS/FreshRSS/blob/master/lib/Minz/Dispatcher.php#L152

**Comment by Frenzie** [comment]
@kevinpapst I don't quite understand. I don't think I wrote that?

**Comment by kevinpapst** [comment]
True, I know. I just ran into that uppercase-vs-lowercase issue and thought when you are working on improving the english docu you could change that one character as well ... seemed easier than opening a new PR for it :-D

**Comment by Frenzie** [comment]
Where should I look for it later? Same file somewhere?

**Comment by kevinpapst** [comment]
Unfortunately I can't manage to create s GH source link, as it always renders the markdown instead of showing its content, but its  the same file just a bit deeper below your changes.

File: docs/fr/developers/03_Backend/05_Extensions.md [hunk 1]
```diff
@@ -332,4 +332,4 @@ TODO :
 
 ### Écrire le fichier configure.phtml
 
-TODO
+Lorsque vous voulez soutenir les configurations utilisateur ou simplement afficher quelques informations, vous devez créer le fichier `configure.phtml`.
```
Comment (Frenzie) [inline docs/fr/developers/03_Backend/05_Extensions.md]: @Alkarex The English was lying by saying something should be translated from French when it was the reverse. This is my translation for the following, but my French isn't stellar:

> When you want to support user configurations for your extension or simply display some information, you have to create the `configure.phtml` file.

Comment (aledeg) [reply docs/fr/developers/03_Backend/05_Extensions.md]: @Frenzie The following is better:

> Lorsque vous voulez ajouter de la configuration à votre extension ou afficher ses informations, vous devez créer le fichier `configure.phtml`

Comment (Frenzie) [reply docs/fr/developers/03_Backend/05_Extensions.md]: @aledeg Thanks! Btw feel free to comment on the English too; I'll give it another read in a couple of days but it's basically done.

Comment (aledeg) [reply docs/fr/developers/03_Backend/05_Extensions.md]: @Frenzie Sure, but at the moment, it's pretty crazy and I don't have much time to spend on FRSS. It's sad but I am trying to keep an eye on it.

**Comment by Alkarex** [comment]
There is indeed something strange with the casing in the Minz code:

The constant is not used:
https://github.com/FreshRSS/FreshRSS/blob/4e174ed9dd84ce4f8de410baf6a6e0fde9160055/lib/Minz/Dispatcher.php#L13

and the `/controllers/` should probably be with uppercase:
https://github.com/FreshRSS/FreshRSS/blob/4e174ed9dd84ce4f8de410baf6a6e0fde9160055/lib/Minz/Dispatcher.php#L152

Let's address that in another PR.

File: docs/en/developers/03_Backend/05_Extensions.md [hunk 2]
```diff
@@ -296,9 +336,9 @@ In addition, you will have a number of methods directly inherited from `Minz_Ext
 
 - The "getters" first: most are explicit enough not to detail them here - `getName()`, `getEntrypoint()`, `getPath()` (allows you to retrieve the path to your extension), `getAuthor()`, `getDescription()`, `getVersion()`, `getType()`.
 - `getFileUrl($filename, $type)` will return the URL to a file in the `static` directory. The first parameter is the name of the file (without `static /`), the second is the type of file to be used (`css` or` js`).
-- `registerController($base_name)` will tell Minz to take into account the given controller in the routing system. The controller must be located in your `Controllers` directory, the name of the file must be` <base_name>Controller.php` and the name of the `FreshExtension_<base_name>_Controller` class.
+- `registerController($base_name)` will tell Minz to take into account the given controller in the routing system. The controller must be located in your `controllers` directory, the name of the file must be` <base_name>Controller.php` and the name of the `FreshExtension_<base_name>_Controller` class.
```
Comment (Frenzie) [inline docs/en/developers/03_Backend/05_Extensions.md]: @Alkarex So you'd like me to remove this diff from the PR?

Comment (Alkarex) [reply docs/en/developers/03_Backend/05_Extensions.md]: Please wait a bit till I find out how it should be :-)

Comment (Alkarex) [reply docs/en/developers/03_Backend/05_Extensions.md]: Yes, as far as I can see, let's go uppercase: https://github.com/FreshRSS/FreshRSS/pull/1729

Comment (kevinpapst) [reply docs/en/developers/03_Backend/05_Extensions.md]: It depends if you want to document the current state (lowercase) or the upcoming change (uppercase). The change to uppercase will introduce a BC break (but lets discuss that in a #1729).

Comment (Alkarex) [reply docs/en/developers/03_Backend/05_Extensions.md]: The documentation follows the code, i.e. the documentation in master is for master, and in dev for dev. So it should be for the upcoming change :-)

**Comment by Frenzie** [comment]
Oh, I also wanted to make sure the French was correct once we had the case issue sorted. :-)

**Comment by Alkarex** [comment]
Your changes looked good enough, at least 👍

File: docs/en/developers/03_Backend/05_Extensions.md [hunk 3]
```diff
@@ -179,7 +206,21 @@ Minz_Request::bad($feedback_bad, $url_array);
...
+
+The solution is to use the `Minz_Translate` class, which allows dynamic translation of FreshRSS (or any Minz-based application). Before using this module, it is necessary to know where to find the strings to be translated. Each language has its own subdirectory in a parent directory named `i18n`. For example, English language files are located in `i18n/fr/`. There are seven different files:
+
+- `admin.php` for anything related to FreshRSS administration
+- `conf.php` for configuration
+- `feedback.php` contains translations of feedback messages
+- `gen.php` stores what is global to FreshRSS (gen for "general")
+- `index.php` for the main page that lists feeds and the About page
+- `install.php` contains strings related FreshRSS installation
+- `sub.php` for subscription management (sub for :subscription")
```
Comment (Frenzie) [inline docs/en/developers/03_Backend/05_Extensions.md:219]: Drat, missed this typo. (Note to self.)

Comment (Alkarex) [reply docs/en/developers/03_Backend/05_Extensions.md:219]: Arf, new PR :-)

Comment (Alkarex) [reply docs/en/developers/03_Backend/05_Extensions.md:219]: If you send a correction, I will merge to /master without making a new release (as this is only the documentation). I will wait for that before starting 1.9.1-dev.

Comment (Frenzie) [reply docs/en/developers/03_Backend/05_Extensions.md:219]: Unless you'd like to get it fixed ASAP I'll just include it in an upcoming PR for some style/grammar fixes in other parts of the docs. :-)

Comment (Frenzie) [reply docs/en/developers/03_Backend/05_Extensions.md:219]: Then again I can just do it from GH itself in two secs.

**Pull request merged**